tests/integration-service/integration-with-env.go

```diff
@@ -0,0 +1,170 @@
+			integrationTestScenario, err = f.AsKubeAdmin.IntegrationController.CreateIntegrationTestScenario("", applicationName, testNamespace, gitURL, itsRevision, pathIntegrationPipelineWithEnv)
+			Expect(err).ShouldNot(HaveOccurred())
+		})
+
+		AfterAll(func() {
+			if !CurrentSpecReport().Failed() {
+				Expect(f.SandboxController.DeleteUserSignup(f.UserName)).To(BeTrue())
+			}
+		})
+
+		When("a new Component is created", func() {
+			It("triggers a build PipelineRun", Label("integration-service"), func() {
+				pipelineRun, err = f.AsKubeDeveloper.IntegrationController.GetBuildPipelineRun(componentName, applicationName, testNamespace, false, "")
+				Expect(err).ShouldNot(HaveOccurred())
+			})
+
+			It("verifies if the build PipelineRun contains the finalizer", Label("integration-service"), func() {
+				Eventually(func() error {
+					pipelineRun, err = f.AsKubeDeveloper.IntegrationController.GetBuildPipelineRun(componentName, applicationName, testNamespace, false, "")
+					Expect(err).ShouldNot(HaveOccurred())
+					if !controllerutil.ContainsFinalizer(pipelineRun, pipelinerunFinalizerByIntegrationService) {
+						return fmt.Errorf("build pipelineRun %s/%s doesn't contain the finalizer: %s yet", pipelineRun.GetNamespace(), pipelineRun.GetName(), pipelinerunFinalizerByIntegrationService)
+					}
+					return nil
+				}, 1*time.Minute, 1*time.Second).Should(Succeed(), "timeout when waiting for finalizer to be added")
+			})
+
+			It("waits for build PipelineRun to succeed", Label("integration-service"), func() {
+				Expect(pipelineRun.Annotations[snapshotAnnotation]).To(Equal(""))
+				Expect(f.AsKubeDeveloper.HasController.WaitForComponentPipelineToBeFinished(originalComponent, "",
+					f.AsKubeAdmin.TektonController, &has.RetryOptions{Retries: 2, Always: true}, pipelineRun)).To(Succeed())
+			})
+		})
+
+		When("the build pipelineRun run succeeded", func() {
+			It("checks if the BuildPipelineRun have the annotation of chains signed", func() {
+				Expect(f.AsKubeDeveloper.IntegrationController.WaitForBuildPipelineRunToGetAnnotated(testNamespace, applicationName, componentName, chainsSignedAnnotation)).To(Succeed())
+			})
+
+			It("checks if the Snapshot is created", func() {
+				snapshot, err = f.AsKubeDeveloper.IntegrationController.WaitForSnapshotToGetCreated("", "", componentName, testNamespace)
+
+			})
+
+			It("checks if the Build PipelineRun got annotated with Snapshot name", func() {
+				Expect(f.AsKubeDeveloper.IntegrationController.WaitForBuildPipelineRunToGetAnnotated(testNamespace, applicationName, componentName, snapshotAnnotation)).To(Succeed())
+			})
+
+			It("verifies that the finalizer has been removed from the build pipelinerun", func() {
+				timeout := "60s"
+				interval := "1s"
+				Eventually(func() error {
+					pipelineRun, err = f.AsKubeDeveloper.IntegrationController.GetBuildPipelineRun(componentName, applicationName, testNamespace, false, "")
```

Comment:
Should we add code in case sometimes the build PLR has been deleted when we try to get it here once the finalizer is removed from it?

Reply:
then it will fail on the line after since the `err != nil` , isn't ?

Reply:
If the finalizer is removed from build PLR, then PLR is deleted quickly, it is expected then the case should not fail. Maybe they don't run so fast.

Reply:
if a Finalizer is removed from PLR this does not means that PLR will be deleted immediately, 
unless there was a request to delete this PLR , isn't ?

Reply:
The pruner jobs that have been added to Konflux are pretty agressive about removing pipelineRuns as soon as they're stored in the Tekton results, so it's not out of the question that a pipelineRun CR will get removed very soon after removing the finalizer.

Reply:
@psturc , wdyt?

Reply:
The case will be more stable if we also consider non-found PLR as pass here because it also indicates the finalizer has been removed.
We may keep it and then update it in the future if it fails.